src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,41 +1,65 @@
-     * #дата_1# - #дата_2# Имя сотрудника - сумма заработанных средств за этот период
-     *
+     * вычисляющий зарплату сотрудников. На вход методу подаётся 2 массива и 2 даты, определяющие
+     * период за который надо вычислить зарплату, первый массив содержит имена сотрудников
+     * организации, второй массив информацию о рабочих часах и ставке. Формат данных второго массива
+     * следующий: дата, имя сотрудника, количество отработанных часов, ставка за 1 час. Метод должен
+     * вернуть отчёт за период, который передали в метод (обе даты включительно) составленный по
+     * следующей форме: Отчёт за период #дата_1# - #дата_2# Имя сотрудника - сумма заработанных
+     * средств за этот период
+     * <p>
      * Пример ввода: date from = 01.04.2019 date to = 30.04.2019
-     *
-     * names:
-     * Сергей
-     * Андрей
-     * София
-     *
-     * data:
-     * 26.04.2019 Сергей 60 50
-     * 26.04.2019 Андрей 3 200
-     * 26.04.2019 Сергей 7 100
-     * 26.04.2019 София 9 100
-     * 26.04.2019 Сергей 11 50
-     * 26.04.2019 Андрей 3 200
-     * 26.04.2019 Сергей 7 100
-     * 26.04.2019 София 9 100
-     * 26.04.2019 Сергей 11 50
-     *
-     * Пример вывода:
-     * Отчёт за период 01.04.2019  - 30.04.2019
-     * Сергей - 1550
-     * Андрей - 600
-     * София - 900
+     * <p>
+     * names: Сергей Андрей София
+     * <p>
+     * data: 26.04.2019 Сергей 60 50 26.04.2019 Андрей 3 200 26.04.2019 Сергей 7 100 26.04.2019
+     * София 9 100 26.04.2019 Сергей 11 50 26.04.2019 Андрей 3 200 26.04.2019 Сергей 7 100
+     * 26.04.2019 София 9 100 26.04.2019 Сергей 11 50
+     * <p>
+     * Пример вывода: Отчёт за период 01.04.2019  - 30.04.2019 Сергей - 1550 Андрей - 600 София -
+     * 900
      */
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder result = new StringBuilder();
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+        LocalDate from = LocalDate.parse(dateFrom, formatter);
+        LocalDate to = LocalDate.parse(dateTo, formatter);
+
+        if (to.isBefore(from)) {
+            return null;
+        }
+        result.append("Отчёт за период ").append(dateFrom).append(" - ").append(dateTo)
+            .append("\n");
+
+        for (String name : names) {
+            int amount = 0;
+            for (String infoAboutEmployee : data) {
+                String[] parsedData = infoAboutEmployee.split(" ");
+                if (name.equals(parsedData[NAME_POSITION])) {
+                    LocalDate reportingDate = LocalDate
+                            .parse(parsedData[REPORTING_DATE_POSITION], formatter);
+                    if (reportingDate.isAfter(from.minusDays(1)) && reportingDate
+                            .isBefore(to.plusDays(1))) {
+                        int hours = Integer.parseInt(parsedData[HOURLY_RATE_POSITION]);
```

Comment:
do we ned this variable? do we use it twice?
the same about rate)

Reply:
Yes, of course. These variables are used in a loop for each line of the report.

Reply:
I think it looks good)
`amount += Integer.parseInt(parsedData[HOURLY_RATE_POSITION]) 
        +Integer.parseInt(parsedData[WORKED_HOURS_POSITION])`